transform/transformers/organisation.py

```diff
@@ -0,0 +1,161 @@
+                ]
+            )
+        )
+
+    @property
+    def cols_to_add(self) -> None:
+        """Add those columns to the dataframe"""
+        return None
+
+    @property
+    def cols_to_drop(self) -> tuple[str, ...]:
+        """Drop those columns from the dataframe"""
+        return (
+            "alternativenames",
+            "country",
+            "pid",
+        )
+
+    @property
+    def cols_to_rename(self) -> dict[str, str]:
+        """Columns to rename. Keys are mapped to the values"""
+        return {
+            "legalname": LEGAL_NAME,
+            "legalshortname": LEGAL_SHORT_NAME,
+            "websiteurl": URL,
+        }
+
+    @staticmethod
+    def harvest_alternative_names(df: DataFrame, harvested_properties: dict) -> None:
+        """
+        Harvest alternative names from specified DataFrame columns and store them as sets in the output dictionary.
+
+        Args:
+            df (DataFrame): The input DataFrame containing columns for alternative names.
+            harvested_properties (Dict[str, List[List[str]]]): A dictionary to store harvested properties.
+
+        Returns:
+            None: The function operates in-place, updating the `harvested_properties` dictionary.
+        """
+        alternative_names_list = df.select(ALTERNATIVENAMES).collect()
+        legal_names_list = df.select(LEGAL_NAME).collect()
+        legal_short_names_list = df.select(LEGAL_SHORT_NAME).collect()
+
+        alternative_names_column = []
+
+        for alternative_name_row, legal_name_row, legal_short_name_row in zip(
+            alternative_names_list, legal_names_list, legal_short_names_list
+        ):
+            alternative_names_set = {
+                legal_name_row[LEGAL_NAME],
+                legal_short_name_row[LEGAL_SHORT_NAME],
+            }
+            if (
+                alternative_name_row[ALTERNATIVENAMES]
+                and len(alternative_name_row[ALTERNATIVENAMES]) > 0
+            ):
+                alternative_names_set.update(alternative_name_row[ALTERNATIVENAMES])
+
+            alternative_names_set = sorted(filter(None, alternative_names_set))
+
+            alternative_names_column.append(list(alternative_names_set))
+
+        harvested_properties[ALTERNATIVE_NAMES] = alternative_names_column
+
+    @staticmethod
+    def harvest_country(df: DataFrame, harvested_properties: dict) -> None:
+        """Harvest country from country.element.code as string"""
+        countries_list = df.select(COUNTRY).collect()
+        country_column = []
+
+        for countries in countries_list:
+            country_raw_val = countries[COUNTRY] or None
+            country_val = country_raw_val["label"] if country_raw_val else None
+            country_column.append(country_val)
+
+        harvested_properties[COUNTRY] = country_column
+
+    @staticmethod
+    def harvest_pids(df: DataFrame, harvested_properties: dict) -> None:
```

Comment:
Pids should have exactly the same schema as everywhere else. E.g.:  
`"pids":"{\"arXiv\": [], \"doi\": [], \"handle\": [], \"pdb\": [], \"pmc\": [], \"pmid\": [], \"w3id\": []}",`
Do organisations have different PIDs?

Reply:
I see, organisations have different PIDs. So what we should do is to create a schema for them (in the same convention). Please check what are all possible values and create a dict where keys are those values. Values: arr<str>

Reply:
Okey, so there are 19 different types of pids in organisations, that's why I created something similar to our pids in research products but I didn't specify the types. If we want to create similar structure my proposal is that it will include `ROR, GRID, ISNI, Wikidata, OrgRef, PIC, FundRef and OrgReg`. It will be like this `"pids":"{\"FundRef\": [], \"GRID\": [], \"ISNI\": [], \"OrgRef\": [], \"OrgReg\": [], \"PIC\": [], \"ROR\": [], \"Wikidata\": []}",`

| pid_type  | value_count | percentage |
|-----------|-------------|------------|
| ROR       | 21609       | 59.53      |
| GRID      | 20644       | 56.85      |
| ISNI      | 16260       | 44.80      |
| Wikidata  | 15167       | 41.80      |
| OrgRef    | 8351        | 23.02      |
| PIC       | 6576        | 18.10      |
| FundRef   | 6197        | 17.05      |
| OrgReg    | 2424        | 6.68       |
| RNSR      | 408         | 1.12       |
| NIF       | 382         | 1.05       |
| RRID      | 331         | 0.91       |
| CNRS      | 324         | 0.89       |
| RingGold  | 299         | 0.82       |
| UKPRN     | 150         | 0.41       |
| HESA      | 147         | 0.40       |
| UCAS      | 137         | 0.38       |
| DGEEC     | 26          | 0.07       |
| VIAF      | 19          | 0.05       |
| LinkedIn  | 1           | 0.00       |

Reply:
Thanks for providing such a nice table! 👍
 I see, there are many keys. So lets create pids in such a schema:
`"pids":"{\"ROR\": ['abc', 'def'], \"Wikidata\": ['gh']}"`
So only not empty keys should be present

Reply:
Also lowercase the keys!

Reply:
Done